pkg/operator/certrotation/signer.go

```diff
@@ -100,7 +101,16 @@ func (c RotatedSigningCASecret) EnsureSigningCertKeyPair(ctx context.Context) (*
 	}
 
 	if modified {
-		actualSigningCertKeyPairSecret, _, err := resourceapply.ApplySecret(ctx, c.Client, c.EventRecorder, signingCertKeyPairSecret)
+		var actualSigningCertKeyPairSecret *corev1.Secret
+		var action string
+		if !exists {
+			actualSigningCertKeyPairSecret, err = c.Client.Secrets(c.Namespace).Create(ctx, signingCertKeyPairSecret, metav1.CreateOptions{})
+			action = "Create"
+		} else {
+			actualSigningCertKeyPairSecret, err = c.Client.Secrets(c.Namespace).Update(ctx, signingCertKeyPairSecret, metav1.UpdateOptions{})
+			action = "Update"
+		}
+		emitEvent(c.EventRecorder, "Secret", action, resourcehelper.FormatResourceForCLIWithNamespace(actualSigningCertKeyPairSecret), err)
```

Comment:
If you do https://github.com/openshift/library-go/pull/1772/files#r1714300480, this would get shorter:

```go
var actualSigningCertKeyPairSecret *corev1.Secret
if !exists {
	actualSigningCertKeyPairSecret, err = c.Client.Secrets(c.Namespace).Create(ctx, signingCertKeyPairSecret, metav1.CreateOptions{})
	emitEvent(c.EventRecorder, "Create", actualSigningCertKeyPairSecret, err)
} else {
	actualSigningCertKeyPairSecret, err = c.Client.Secrets(c.Namespace).Update(ctx, signingCertKeyPairSecret, metav1.UpdateOptions{})
	emitEvent(c.EventRecorder, "Update", actualSigningCertKeyPairSecret, err)
}

```

Reply:
to minimize copying important params I introduced "action" which is the only variable there